File: python-project-euler-100/p005.py
import time
from Solution import Solution

class P005(Solution):
    def solve(self):
        self.problem_number = 5
        last = 20
        sequence = []
        result = 1
        number = 2
        counter = number
        while(counter <= last):
            for i in sequence:
                if number % i == 0:
                    number //= i
            result *= number
            sequence.append(number)
            counter += 1
            number = counter
        
        return result

def main():
	P005().run()

if __name__ == "__main__":
	main()

File: typescript-project-euler-100/src/p005.ts
import { Solution } from "./Solution";

export class P005 extends Solution{
    solve() {
        const last = 20;
		const sequence: number[] = [];
		let result = 1;
		let number = 2;
		let counter = number;
		while (counter <= last) {
            let i: number;
            let j: number;
			for (i = 0; i < sequence.length; i++) {
                j = sequence[i];
				if (number % j === 0) {
                    number = number / j;
				}
			}

			result *= number;
			sequence.push(number);
			counter += 1;
			number = counter;
		}

        return result;
    }
}
